rclcpp_components/src/component_manager.cpp

```diff
@@ -46,6 +46,20 @@ ComponentManager::ComponentManager(
   listNodes_srv_ = create_service<ListNodes>(
     "~/_container/list_nodes",
     std::bind(&ComponentManager::on_list_nodes, this, _1, _2, _3));
+
+  {
+    rcl_interfaces::msg::ParameterDescriptor desc{};
+    desc.name = "thread_num";
+    desc.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
```

Comment:
name and type field shouldn't be set manually in the descriptor, `declare_parameter()` ignores them

Reply:
Fixed at https://github.com/ros2/rclcpp/pull/1708/commits/06c5651caf1864aae7e74a8207aa0b180651b6ec.